src/backend/src/services/teams.services.ts

```diff
@@ -379,14 +387,23 @@ export default class TeamsService {
       throw new HttpException(400, 'Cannot create a teamType with a name that already exists');
     }
 
-    const teamType = await prisma.team_Type.create({
+    let teamType = await prisma.team_Type.create({
```

Comment:
You don't have to change this to a let if you put `if (!calendarId) const calendarId = await createCalendar(teamType.name)` above this 
Then, remove the duplicate code below